openwisp_monitoring/device/base/models.py

```diff
@@ -454,6 +456,23 @@ def handle_disabled_organization(cls, organization_id):
             status='unknown'
         )
 
+    @classmethod
+    def _get_critical_metric_keys(cls):
+        return [metric['key'] for metric in get_critical_device_metrics()]
+
+    @classmethod
+    def handle_critical_metric(cls, instance, **kwargs):
+        critical_metrics = cls._get_critical_metric_keys()
+        if instance.check_type in critical_metrics:
+            try:
+                device_monitoring = cls.objects.get(device=instance.content_object)
+                if not instance.is_active or kwargs.get('signal') == post_delete:
+                    device_monitoring.update_status('unknown')
+                elif kwargs.get('signal') == post_save and instance.is_active:
+                    device_monitoring.update_status('ok')
```

Comment:
This is wrong. Only checks are responsible for knowing whether a device is ok, we can't know without checking.
Was it specified anywhere that we should set the status to ok when the change goes back to active? Why did you add this?